### PR TITLE
Update dependencies, add new supported framework identifiers, tune CI

### PR DESCRIPTION
**Pull request by halex2005**
- update nunit dependency to v3.10.1
- update NUnit3TestAdapter dependency to v3.10.0
- update Microsoft.NET.Test.Sdk dependency to v15.6.1
- add new `--framework` supported parameters:
    + .NET Framework versions (net35, net40, net45, net451, net452, net46, net461, net462, net47, net471)
    + .NET Core 2.1 (netcoreapp2.1)
- add CI using travis-ci (linux, mac osx) and appveyor-ci (windows)

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -1,3 +1,14 @@
+19 March 2018, v1.4.0
+---------------------
+
+- update nunit dependency to v3.10.0
+- update NUnit3TestAdapter dependency to v3.10.0
+- update Microsoft.NET.Test.Sdk dependency to v15.6.1
+- add new `--framework` supported parameters:
+    + .NET Framework versions (net40, net45, net451, net452, net46, net461, net462, net47, net471)
```
Comment (ChrisMaddock) [inline changelog.md]: It looks like the net40 target is listed here, but not actually supported? Unless I'm misreading these commits?

(FWIW - I'd love it to be added if there's not additional cost, I work in .NET 4.0 on a day-to-day basis. 😄)

File: nunit-template.nuspec [hunk 2]
```diff
@@ -16,9 +16,12 @@
...
     </packageTypes>
     <releaseNotes>
-- use nunit v3.9.0
-- use NUnit3TestAdapter v3.9.0
-- use Microsoft.NET.Test.Sdk v15.5.0
+- update nunit dependency to v3.10.0
+- update NUnit3TestAdapter dependency to v3.10.0
+- update Microsoft.NET.Test.Sdk dependency to v15.6.0
+- add new `--framework` supported parameters:
+    + .NET Framework versions (net40, net45, net451, net452, net46, net461, net462, net47, net471)
```
Comment (ChrisMaddock) [inline nunit-template.nuspec]: As above re: net40

File: Content/dotnet-new-nunit-fsharp/Company.TestProject1.fsproj [hunk 3]
```diff
@@ -9,9 +9,9 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="nunit" Version="3.9.0" />
-    <PackageReference Include="NUnit3TestAdapter" Version="3.9.0" />
-    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.5.0" />
+    <PackageReference Include="nunit" Version="3.10.0" />
```
Comment (ChrisMaddock) [inline Content/dotnet-new-nunit-fsharp/Company.TestProject1.fsproj]: NUnit 3.10.0 had a critical bug - I'd recommend going straight to 3.10.1. 😄

Comment (halex2005) [reply Content/dotnet-new-nunit-fsharp/Company.TestProject1.fsproj]: thanks! done)

File: nunit-template.nuspec (hunk 2 above)
Comment (halex2005) [reply nunit-template.nuspec]: fixed

File: changelog.md (hunk 1 above)
Comment (halex2005) [reply changelog.md]: Thanks for feedback! Yes, you right about net40. I'm have returned back it now)

Comment (ChrisMaddock) [reply changelog.md]: Awesome - thanks!

**Pull request merged**